module/validation/receipt_validator.go

```diff
@@ -376,15 +370,8 @@ func (v *receiptValidator) validateReceipt(receipt *flow.ExecutionReceiptMeta, b
 
 // check the receipt's data integrity by checking its result has
 // both final statecommitment and initial statecommitment
-func IntegrityCheck(receipt *flow.ExecutionReceipt) (flow.StateCommitment, flow.StateCommitment, error) {
-	final, ok := receipt.ExecutionResult.FinalStateCommitment()
-	if !ok {
-		return nil, nil, fmt.Errorf("execution receipt without FinalStateCommit: %x", receipt.ID())
-	}
-
-	init, ok := receipt.ExecutionResult.InitialStateCommit()
-	if !ok {
-		return nil, nil, fmt.Errorf("execution receipt without InitialStateCommit: %x", receipt.ID())
-	}
-	return init, final, nil
+func IntegrityCheck(receipt *flow.ExecutionReceipt) (flow.StateCommitment, flow.StateCommitment) {
```

Comment:
shall we have checks on if it is returning the EmptyStateCommitment?

Reply:
This function is getting refactored, there will be a further check for an empty block rather than a check for `EmptyStateCommitment`.